core-bundle/contao/drivers/DC_Table.php

```diff
@@ -2215,10 +2215,11 @@ public function edit($intId=null, $ajaxId=null)
 
 			if (!Input::get('nc'))
 			{
-				if (!($GLOBALS['TL_DCA'][$this->strTable]['config']['closed'] ?? null) && !($GLOBALS['TL_DCA'][$this->strTable]['config']['notCreatable'] ?? null))
+				if (!($GLOBALS['TL_DCA'][$this->strTable]['config']['closed'] ?? null) && !($GLOBALS['TL_DCA'][$this->strTable]['config']['notCreatable'] ?? null) && $security->isGranted(ContaoCorePermissions::DC_PREFIX . $this->strTable, new CreateAction($this->strTable, array('pid' => $this->intCurrentPid))))
 				{
 					$arrButtons['saveNcreate'] = '<button type="submit" name="saveNcreate" id="saveNcreate" class="tl_submit" accesskey="n">' . $GLOBALS['TL_LANG']['MSC']['saveNcreate'] . '</button>';
 
+					// If generic CreateAction is granted and we're editing a record, we already know we have permissions to copy the current record
```

Comment:
I don‘t understand this comment. What is it referring to?

Reply:
it means we don't need to check permissions again here. A "copy" permission consists of checking whether the current record can be read and a new record can be written. Since we already are in DC_Table::edit() method, the read action (and UpdateAction) is already checked.